cmd/juju/commands/ssh_common.go

```diff
@@ -392,10 +394,22 @@ func (c *SSHCommon) reachableAddressGetter(entity string) (string, error) {
 	} else if len(addresses) == 0 {
 		return "", network.NoAddressError("available")
 	}
+	publicKeys := []string{}
+	if !c.noHostKeyChecks {
+		publicKeys, err = c.apiClient.PublicKeys(entity)
+		if err != nil {
+			// We ignore NotFound errors, as we may not have finished registering
```

Comment:
IIRC, the issue was that a test was expecting a particular context around what entity we were missing keys for.  I can fix that by just changing the returned error instead of skipping NotFound.